app/modules/finance/operations_router.py

```
@@ -48,13 +52,35 @@ def create_payment_voucher(payload:dict,db:Session=Depends(get_db),principal:Pri
+    if not r: raise HTTPException(404,'Payment voucher not found')
+    required=_required_approvals(_money(r.amount)); approvals=db.query(m.FinanceApproval).filter_by(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id,decision='APPROVE').order_by(m.FinanceApproval.sequence).all()
+    return {'voucher_id':r.id,'voucher_number':r.voucher_number,'amount':_money(r.amount),'required_approvals':required,'approved_count':len(approvals),'next_sequence':len(approvals)+1 if len(approvals)<required else None,'status':r.status,'approvals':[{'sequence':a.sequence,'approver':a.approver,'decided_at':a.decided_at} for a in approvals]}
 @router.post('/finance/payment-vouchers/{voucher_id}/approve')
 def approve_payment_voucher(voucher_id:int,sequence:int=Query(1,ge=1,le=3),decision:str=Query('APPROVE'),db:Session=Depends(get_db),principal:Principal=Depends(require_role('admin'))):
     r=db.query(m.FinancePaymentVoucher).filter_by(id=voucher_id,school_id=principal.school_id).first()
     if not r:raise HTTPException(404,'Payment voucher not found')
     if r.requested_by==principal.user_id:raise HTTPException(403,'Requester cannot approve their own voucher')
     if decision not in {'APPROVE','REJECT'}:raise HTTPException(400,'decision must be APPROVE or REJECT')
-    db.add(m.FinanceApproval(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id,sequence=sequence,required_role='admin',approver=principal.user_id,decision=decision,amount_threshold=r.amount,reason='Voucher approval',decided_at=datetime.now(timezone.utc)));r.status='APPROVED' if decision=='APPROVE' else 'REJECTED';_audit(db,principal,'approve','payment_voucher',r.id,f'Voucher {r.voucher_number}: {decision}');db.commit();db.refresh(r);return r
+    if r.status in {'REJECTED','PAID'}:raise HTTPException(409,f'Voucher is already {r.status.lower()}')
+    required=_required_approvals(_money(r.amount)); existing=db.query(m.FinanceApproval).filter_by(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id).all()
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)